### PR TITLE
RIGs module re-haul and re-balance

### PR DESCRIPTION
**Pull request by Sindorman**
As discussed with @Arrow768 this PR makes each module have a category, and each rig have a list of allowed categories. Modules of not allowed category cannot be installed, unless they were pre-installed.

**feedback thread:** https://forums.aurorastation.org/topic/11396-rigs-re-haul-and-re-balance/

- Module categories are "general", "light combat", "heavy combat", "special", "medical", "utility", "vaurca"
### RIGS re-balance
ERT rig armour change:
commander: armor = melee = 60, bullet = 50, laser = 30, energy = 15, bomb = 30, bio = 100, rad = 100
security:  melee = 65, bullet = 55, laser = 40, energy = 15, bomb = 30, bio = 100, rad = 80
medical:  melee = 55, bullet = 45, laser = 30, energy = 15, bomb = 30, bio = 100, rad = 100
engineering:  melee = 55, bullet = 45, laser = 30, energy = 15, bomb = 30, bio = 100, rad = 100

- Industrial rig slowdown will be decreased from 3 to 2. 

- Medical rig slowdown will be decreased from 1 to 0

### Modules categories are:

- general: maneuvering jets, IIS module, actuators, sign fabricator, matter decompiler, cleaner grenade launcher, cooling unit, pen, stamp, paper dispenser, hardsuit visors such as mesons, material, individually or combined sometimes.

- light combat: flash, energy gun, taser, thermal visor, night visor, sechug visor.

- heavy combat: grenade launcher, mounted laser/cannon, combat actuators, (combat) chemical dispenser.

- medical: medical visor, medical chem injector, medical health scanner.

- utility: any kind of drills, mounted plasma cutter, mounted thermal drill, ore scanner, anomaly scanner, RCD.

- special: metal stars fabricator, datajack, electrowarfare module, power sink, stealth field, teleporter(blink), energy net, self destruct, emp shielding, emergency power generator, emag hand, voice synthesizer, ninja chemical dispenser, all-in-one visor.

- vaurca: vaurca chemical dispenser, burrowing lasers, neural lattice

### Rigs categories change

- ERT rigs are more specialized. ERT security rig can only have "general", "light combat", "heavy combat", "special" modules. ERT medical rig can only have "general", "light combat", "heavy combat", "special", "medical" modules. ERT engineering rig can only have "general", "light combat", "heavy combat", "special", "utility" modules. ERT command rig can accept any modules.

- Every RIG can support general category

- HAPT, ninja, merc, terminator RIGs can accept any modules.

- Varuca Rigs can only accept "general", "light combat", "heavy combat", "special", "vaurca" modules

- TCFL rigs can only accept "general", "light combat", "heavy combat", "special", "medical", "utility" modules

- Other rigs can accept modules as category suggests. Medical - general and medical. CE/EVA/Industrial RIGs - general, utility. AMI - general, utility. hazard hardsuit - general, light combat, and heavy combat.

- Hacker and stealth RIGs - general, light combat combat, special modules

TODO:

- [x] Testing

### CODE REVIEW

**Comment by VTCobaltblood** [comment]
What about the AMI and its counter module?

**Comment by Fire-and-Glory** [comment]
Making visors and HUDs be general items instead of specialized would make more sense, imo.

**Comment by LordFowl** [comment]
This will definitely require a feedback thread.

File: code/modules/clothing/spacesuits/rig/suits/ert.dm [hunk 1]
```diff
@@ -26,14 +27,18 @@
 		/obj/item/rig_module/datajack,
 		/obj/item/rig_module/actuators/combat
 		)
+	
+	allowed_module_types = MODULE_GENERAL | MODULE_LIGHT_COMBAT | MODULE_HEAVY_COMBAT | MODULE_SPECIAL | MODULE_MEDICAL | MODULE_UTILITY | MODULE_VAURCA
 
 /obj/item/weapon/rig/ert/engineer
 	name = "ERT-E suit control module"
-	desc = "A suit worn by the engineering division of an Emergency Response Team. Has orange highlights. Armoured and space ready."
+	desc = "A suit worn by the engineering division of an Emergency Response Team. Has orange highlights. Less armoured than security varient, but offers full radiation protection."
```
Comment (Arrow768) [inline code/modules/clothing/spacesuits/rig/suits/ert.dm]:
```suggestion
	desc = "A suit worn by the engineering division of an Emergency Response Team. Has orange highlights. Less armoured than the security variant, but offers full radiation protection."
```

File: code/modules/clothing/spacesuits/rig/suits/ert.dm [hunk 2]
```diff
@@ -44,14 +49,17 @@
 		/obj/item/rig_module/device/rcd,
 		/obj/item/rig_module/actuators
 		)
+	allowed_module_types = MODULE_GENERAL | MODULE_LIGHT_COMBAT | MODULE_HEAVY_COMBAT | MODULE_SPECIAL | MODULE_UTILITY 
 
 /obj/item/weapon/rig/ert/medical
 	name = "ERT-M suit control module"
-	desc = "A suit worn by the medical division of an Emergency Response Team. Has white highlights. Armoured and space ready."
+	desc = "A suit worn by the medical division of an Emergency Response Team. Has white highlights. Less armoured than security varient, but offers full radiation protection."
```
Comment (Arrow768) [inline code/modules/clothing/spacesuits/rig/suits/ert.dm]:
```suggestion
	desc = "A suit worn by the medical division of an Emergency Response Team. Has white highlights. Less armoured than the security variant, but offers full radiation protection."
```

File: code/modules/clothing/spacesuits/rig/suits/ert.dm [hunk 3]
```diff
@@ -76,10 +88,11 @@
 		/obj/item/rig_module/mounted/egun,
 		/obj/item/rig_module/actuators
 		)
+	allowed_module_types = MODULE_GENERAL | MODULE_LIGHT_COMBAT | MODULE_HEAVY_COMBAT | MODULE_SPECIAL
 
 /obj/item/weapon/rig/ert/janitor
 	name = "ERT-J suit control module"
-	desc = "A suit worn by the janitoral division of an Emergency Response Team. Has purple highlights. Armoured and space ready."
+	desc = "A suit worn by the janitoral division of an Emergency Response Team. Has purple highlights. Less armoured than security varient, but offers full radiation protection."
```
Comment (Arrow768) [inline code/modules/clothing/spacesuits/rig/suits/ert.dm]:
```suggestion
	desc = "A suit worn by the janitoral division of an Emergency Response Team. Has purple highlights. Less armoured than security the variant, but offers full radiation protection."
```

**Comment by Sindorman** [comment]
DNM until #6120 is finished, reviewed and approved.

**Pull request merged**